src/structured-text/parser.y

```diff
@@ -0,0 +1,701 @@
+
+#ifdef STRUCTURED_TEXT_DEBUG
+#define YYDEBUG 1
+#endif
+#define PARSER structured_text_parser
+
+#include "structured_text_parser.h"
+#include "converters/convert_string_value.h"
+#include <util/std_expr.h>
+
+int yystructured_textlex();
+extern char *yystructured_texttext;
+
+#define YYSTYPE unsigned
+#define YYSTYPE_IS_TRIVIAL 1
+
+#include "structured_text_y.tab.h"
+
+// Visual Studio
+#ifdef _MSC_VER
+// Disable warnings for possible loss of data.
+#pragma warning(disable:4242)
+#pragma warning(disable:4244)
+// Disable warning for signed/unsigned mismatch.
+#pragma warning(disable:4365)
+// Disable warning for switch with default but no case labels.
+#pragma warning(disable:4065)
+// Disable warning for unreachable code.
+#pragma warning(disable:4702)
+#endif
+
+/*** Token declaration *******************************************************/
+%}
+
+/*** STL file structure keywords *********************************************/
+%token TOK_VERSION 		   		"VERSION"
+%token TOK_BEGIN   		   		"BEGIN"
+%token TOK_FUNCTION_BLOCK  		"FUNCTION_BLOCK"
+%token TOK_END_FUNCTION_BLOCK	"END_FUNCTION_BLOCK"
+%token TOK_FUNCTION				"FUNCTION"
+%token TOK_END_FUNCTION			"END_FUNCTION"
+%token TOK_VAR_INPUT			"VAR_INPUT"
+%token TOK_VAR_OUTPUT			"VAR_OUTPUT"
+%token TOK_END_VAR				"END_VAR"
+%token TOK_NETWORK				"NETWORK"
+%token TOK_TITLE				"TITLE"
+
+/*** Siemens types ***********************************************************/
+%token TOK_INT					"Int"
+%token TOK_DINT					"DInt"
+%token TOK_REAL					"Real"
+%token TOK_BOOL					"Bool"
+%token TOK_VOID					"Void"
+
+/*** Operators ***************************************************************/
+%token TOK_LOAD					"L"
+%token TOK_TRANSFER				"T"
+%token TOK_NOP					"NOP"
+%token TOK_AND					"A"
+%token TOK_AND_NOT				"AN"
+%token TOK_OR					"O"
+%token TOK_OR_NOT				"ON"
+%token TOK_XOR					"X"
+%token TOK_XOR_NOT				"XN"
+%token TOK_CONST_ADD			"+"
+%token TOK_ACCU_INT_ADD			"+I"
+%token TOK_ACCU_INT_SUB			"-I"
+%token TOK_ACCU_INT_MUL			"*I"
+%token TOK_ACCU_INT_DIV			"/I"
+%token TOK_ACCU_REAL_ADD		"+R"
+%token TOK_ACCU_REAL_SUB		"-R"
+%token TOK_ACCU_REAL_MUL		"*R"
+%token TOK_ACCU_REAL_DIV		"/R"
+%token TOK_ACCU_DINT_ADD		"+D"
+%token TOK_ACCU_DINT_SUB		"-D"
+%token TOK_ACCU_DINT_MUL		"*D"
+%token TOK_ACCU_DINT_DIV		"/D"
+%token TOK_ASSIGNMENT			":="
+
+/*** Value tokens ***/
+%token TOK_INT_LITERAL
+%token TOK_BOOL_LITERAL
+%token TOK_REAL_LITERAL
+%token TOK_IDENTIFIER
+%token TOK_TITLE_VALUE
+%token TOK_LABEL
+
+/*** Priority, associativity, etc. definitions *******************************/
+
+%start init
+
+%{
+/*** Grammar rules ***********************************************************/
+
+// The follwing abbreviations will be used:
+//   Zom: "Zero or more", eqivalent to the '*' quantifier
+//   Opt: "Optional", equivalent to the '?' quantifier
+//   Oom: "One or more", equivalent to the '+' quantifier
+%}
+%%
+
+init:
+	init FB_Decl
+	| init Func_Decl
+	| /* nothing */
+	;
```

Comment:
Nit picking: most rules appear to use 4-character indent, but a few others use 8 characters or so?!